// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: pb/secret.v1.proto

package pb

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on ListSecretsRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ListSecretsRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ListSecretsRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ListSecretsRequestMultiError, or nil if none found.
func (m *ListSecretsRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ListSecretsRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetClusterName()) < 1 {
		err := ListSecretsRequestValidationError{
			field:  "ClusterName",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetNamespace()) < 1 {
		err := ListSecretsRequestValidationError{
			field:  "Namespace",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	// no validation rules for Keyword

	// no validation rules for Page

	// no validation rules for PageSize

	// no validation rules for ListAll

	// no validation rules for Type

	if len(errors) > 0 {
		return ListSecretsRequestMultiError(errors)
	}

	return nil
}

// ListSecretsRequestMultiError is an error wrapping multiple validation errors
// returned by ListSecretsRequest.ValidateAll() if the designated constraints
// aren't met.
type ListSecretsRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ListSecretsRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ListSecretsRequestMultiError) AllErrors() []error { return m }

// ListSecretsRequestValidationError is the validation error returned by
// ListSecretsRequest.Validate if the designated constraints aren't met.
type ListSecretsRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListSecretsRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListSecretsRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListSecretsRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListSecretsRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListSecretsRequestValidationError) ErrorName() string {
	return "ListSecretsRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListSecretsRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListSecretsRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListSecretsRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListSecretsRequestValidationError{}

// Validate checks the field values on SecretReference with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *SecretReference) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SecretReference with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// SecretReferenceMultiError, or nil if none found.
func (m *SecretReference) ValidateAll() error {
	return m.validate(true)
}

func (m *SecretReference) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Kind

	// no validation rules for Name

	if len(errors) > 0 {
		return SecretReferenceMultiError(errors)
	}

	return nil
}

// SecretReferenceMultiError is an error wrapping multiple validation errors
// returned by SecretReference.ValidateAll() if the designated constraints
// aren't met.
type SecretReferenceMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SecretReferenceMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SecretReferenceMultiError) AllErrors() []error { return m }

// SecretReferenceValidationError is the validation error returned by
// SecretReference.Validate if the designated constraints aren't met.
type SecretReferenceValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SecretReferenceValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SecretReferenceValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SecretReferenceValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SecretReferenceValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SecretReferenceValidationError) ErrorName() string { return "SecretReferenceValidationError" }

// Error satisfies the builtin error interface
func (e SecretReferenceValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSecretReference.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SecretReferenceValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SecretReferenceValidationError{}

// Validate checks the field values on SecretItem with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *SecretItem) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SecretItem with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in SecretItemMultiError, or
// nil if none found.
func (m *SecretItem) ValidateAll() error {
	return m.validate(true)
}

func (m *SecretItem) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetName()) < 1 {
		err := SecretItemValidationError{
			field:  "Name",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	// no validation rules for RefCount

	// no validation rules for Labels

	// no validation rules for Annotations

	if m.GetCreateTime() < 1 {
		err := SecretItemValidationError{
			field:  "CreateTime",
			reason: "value must be greater than or equal to 1",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if m.GetUpdateTime() < 1 {
		err := SecretItemValidationError{
			field:  "UpdateTime",
			reason: "value must be greater than or equal to 1",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	// no validation rules for Data

	if utf8.RuneCountInString(m.GetYaml()) < 1 {
		err := SecretItemValidationError{
			field:  "Yaml",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	for idx, item := range m.GetRefs() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, SecretItemValidationError{
						field:  fmt.Sprintf("Refs[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, SecretItemValidationError{
						field:  fmt.Sprintf("Refs[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return SecretItemValidationError{
					field:  fmt.Sprintf("Refs[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	// no validation rules for Type

	// no validation rules for TlsCrt

	// no validation rules for TlsKey

	// no validation rules for Registry

	// no validation rules for Username

	// no validation rules for Password

	// no validation rules for IsFollowed

	if len(errors) > 0 {
		return SecretItemMultiError(errors)
	}

	return nil
}

// SecretItemMultiError is an error wrapping multiple validation errors
// returned by SecretItem.ValidateAll() if the designated constraints aren't met.
type SecretItemMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SecretItemMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SecretItemMultiError) AllErrors() []error { return m }

// SecretItemValidationError is the validation error returned by
// SecretItem.Validate if the designated constraints aren't met.
type SecretItemValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SecretItemValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SecretItemValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SecretItemValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SecretItemValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SecretItemValidationError) ErrorName() string { return "SecretItemValidationError" }

// Error satisfies the builtin error interface
func (e SecretItemValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSecretItem.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SecretItemValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SecretItemValidationError{}

// Validate checks the field values on ListSecretsResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ListSecretsResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ListSecretsResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ListSecretsResponseMultiError, or nil if none found.
func (m *ListSecretsResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ListSecretsResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Total

	for idx, item := range m.GetList() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, ListSecretsResponseValidationError{
						field:  fmt.Sprintf("List[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, ListSecretsResponseValidationError{
						field:  fmt.Sprintf("List[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListSecretsResponseValidationError{
					field:  fmt.Sprintf("List[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return ListSecretsResponseMultiError(errors)
	}

	return nil
}

// ListSecretsResponseMultiError is an error wrapping multiple validation
// errors returned by ListSecretsResponse.ValidateAll() if the designated
// constraints aren't met.
type ListSecretsResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ListSecretsResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ListSecretsResponseMultiError) AllErrors() []error { return m }

// ListSecretsResponseValidationError is the validation error returned by
// ListSecretsResponse.Validate if the designated constraints aren't met.
type ListSecretsResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListSecretsResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListSecretsResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListSecretsResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListSecretsResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListSecretsResponseValidationError) ErrorName() string {
	return "ListSecretsResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListSecretsResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListSecretsResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListSecretsResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListSecretsResponseValidationError{}

// Validate checks the field values on CreateOrUpdateSecretByYamlRequest with
// the rules defined in the proto definition for this message. If any rules
// are violated, the first error encountered is returned, or nil if there are
// no violations.
func (m *CreateOrUpdateSecretByYamlRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreateOrUpdateSecretByYamlRequest
// with the rules defined in the proto definition for this message. If any
// rules are violated, the result is a list of violation errors wrapped in
// CreateOrUpdateSecretByYamlRequestMultiError, or nil if none found.
func (m *CreateOrUpdateSecretByYamlRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *CreateOrUpdateSecretByYamlRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetClusterName()) < 1 {
		err := CreateOrUpdateSecretByYamlRequestValidationError{
			field:  "ClusterName",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetYaml()) < 1 {
		err := CreateOrUpdateSecretByYamlRequestValidationError{
			field:  "Yaml",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return CreateOrUpdateSecretByYamlRequestMultiError(errors)
	}

	return nil
}

// CreateOrUpdateSecretByYamlRequestMultiError is an error wrapping multiple
// validation errors returned by
// CreateOrUpdateSecretByYamlRequest.ValidateAll() if the designated
// constraints aren't met.
type CreateOrUpdateSecretByYamlRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreateOrUpdateSecretByYamlRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreateOrUpdateSecretByYamlRequestMultiError) AllErrors() []error { return m }

// CreateOrUpdateSecretByYamlRequestValidationError is the validation error
// returned by CreateOrUpdateSecretByYamlRequest.Validate if the designated
// constraints aren't met.
type CreateOrUpdateSecretByYamlRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreateOrUpdateSecretByYamlRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreateOrUpdateSecretByYamlRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreateOrUpdateSecretByYamlRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreateOrUpdateSecretByYamlRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreateOrUpdateSecretByYamlRequestValidationError) ErrorName() string {
	return "CreateOrUpdateSecretByYamlRequestValidationError"
}

// Error satisfies the builtin error interface
func (e CreateOrUpdateSecretByYamlRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreateOrUpdateSecretByYamlRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreateOrUpdateSecretByYamlRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreateOrUpdateSecretByYamlRequestValidationError{}

// Validate checks the field values on CreateOrUpdateSecretByYamlResponse with
// the rules defined in the proto definition for this message. If any rules
// are violated, the first error encountered is returned, or nil if there are
// no violations.
func (m *CreateOrUpdateSecretByYamlResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreateOrUpdateSecretByYamlResponse
// with the rules defined in the proto definition for this message. If any
// rules are violated, the result is a list of violation errors wrapped in
// CreateOrUpdateSecretByYamlResponseMultiError, or nil if none found.
func (m *CreateOrUpdateSecretByYamlResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *CreateOrUpdateSecretByYamlResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return CreateOrUpdateSecretByYamlResponseMultiError(errors)
	}

	return nil
}

// CreateOrUpdateSecretByYamlResponseMultiError is an error wrapping multiple
// validation errors returned by
// CreateOrUpdateSecretByYamlResponse.ValidateAll() if the designated
// constraints aren't met.
type CreateOrUpdateSecretByYamlResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreateOrUpdateSecretByYamlResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreateOrUpdateSecretByYamlResponseMultiError) AllErrors() []error { return m }

// CreateOrUpdateSecretByYamlResponseValidationError is the validation error
// returned by CreateOrUpdateSecretByYamlResponse.Validate if the designated
// constraints aren't met.
type CreateOrUpdateSecretByYamlResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreateOrUpdateSecretByYamlResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreateOrUpdateSecretByYamlResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreateOrUpdateSecretByYamlResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreateOrUpdateSecretByYamlResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreateOrUpdateSecretByYamlResponseValidationError) ErrorName() string {
	return "CreateOrUpdateSecretByYamlResponseValidationError"
}

// Error satisfies the builtin error interface
func (e CreateOrUpdateSecretByYamlResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreateOrUpdateSecretByYamlResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreateOrUpdateSecretByYamlResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreateOrUpdateSecretByYamlResponseValidationError{}

// Validate checks the field values on DeleteSecretRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *DeleteSecretRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteSecretRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteSecretRequestMultiError, or nil if none found.
func (m *DeleteSecretRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteSecretRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetClusterName()) < 1 {
		err := DeleteSecretRequestValidationError{
			field:  "ClusterName",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetNamespace()) < 1 {
		err := DeleteSecretRequestValidationError{
			field:  "Namespace",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetName()) < 1 {
		err := DeleteSecretRequestValidationError{
			field:  "Name",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return DeleteSecretRequestMultiError(errors)
	}

	return nil
}

// DeleteSecretRequestMultiError is an error wrapping multiple validation
// errors returned by DeleteSecretRequest.ValidateAll() if the designated
// constraints aren't met.
type DeleteSecretRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteSecretRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteSecretRequestMultiError) AllErrors() []error { return m }

// DeleteSecretRequestValidationError is the validation error returned by
// DeleteSecretRequest.Validate if the designated constraints aren't met.
type DeleteSecretRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteSecretRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteSecretRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteSecretRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteSecretRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteSecretRequestValidationError) ErrorName() string {
	return "DeleteSecretRequestValidationError"
}

// Error satisfies the builtin error interface
func (e DeleteSecretRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteSecretRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteSecretRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteSecretRequestValidationError{}

// Validate checks the field values on DeleteSecretResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *DeleteSecretResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteSecretResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteSecretResponseMultiError, or nil if none found.
func (m *DeleteSecretResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteSecretResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return DeleteSecretResponseMultiError(errors)
	}

	return nil
}

// DeleteSecretResponseMultiError is an error wrapping multiple validation
// errors returned by DeleteSecretResponse.ValidateAll() if the designated
// constraints aren't met.
type DeleteSecretResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteSecretResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteSecretResponseMultiError) AllErrors() []error { return m }

// DeleteSecretResponseValidationError is the validation error returned by
// DeleteSecretResponse.Validate if the designated constraints aren't met.
type DeleteSecretResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteSecretResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteSecretResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteSecretResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteSecretResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteSecretResponseValidationError) ErrorName() string {
	return "DeleteSecretResponseValidationError"
}

// Error satisfies the builtin error interface
func (e DeleteSecretResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteSecretResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteSecretResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteSecretResponseValidationError{}

// Validate checks the field values on SecretDetailRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *SecretDetailRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SecretDetailRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// SecretDetailRequestMultiError, or nil if none found.
func (m *SecretDetailRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *SecretDetailRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetClusterName()) < 1 {
		err := SecretDetailRequestValidationError{
			field:  "ClusterName",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetNamespace()) < 1 {
		err := SecretDetailRequestValidationError{
			field:  "Namespace",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetName()) < 1 {
		err := SecretDetailRequestValidationError{
			field:  "Name",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return SecretDetailRequestMultiError(errors)
	}

	return nil
}

// SecretDetailRequestMultiError is an error wrapping multiple validation
// errors returned by SecretDetailRequest.ValidateAll() if the designated
// constraints aren't met.
type SecretDetailRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SecretDetailRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SecretDetailRequestMultiError) AllErrors() []error { return m }

// SecretDetailRequestValidationError is the validation error returned by
// SecretDetailRequest.Validate if the designated constraints aren't met.
type SecretDetailRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SecretDetailRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SecretDetailRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SecretDetailRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SecretDetailRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SecretDetailRequestValidationError) ErrorName() string {
	return "SecretDetailRequestValidationError"
}

// Error satisfies the builtin error interface
func (e SecretDetailRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSecretDetailRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SecretDetailRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SecretDetailRequestValidationError{}

// Validate checks the field values on SecretDetailResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *SecretDetailResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SecretDetailResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// SecretDetailResponseMultiError, or nil if none found.
func (m *SecretDetailResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *SecretDetailResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetDetail()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, SecretDetailResponseValidationError{
					field:  "Detail",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, SecretDetailResponseValidationError{
					field:  "Detail",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetDetail()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return SecretDetailResponseValidationError{
				field:  "Detail",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return SecretDetailResponseMultiError(errors)
	}

	return nil
}

// SecretDetailResponseMultiError is an error wrapping multiple validation
// errors returned by SecretDetailResponse.ValidateAll() if the designated
// constraints aren't met.
type SecretDetailResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SecretDetailResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SecretDetailResponseMultiError) AllErrors() []error { return m }

// SecretDetailResponseValidationError is the validation error returned by
// SecretDetailResponse.Validate if the designated constraints aren't met.
type SecretDetailResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SecretDetailResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SecretDetailResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SecretDetailResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SecretDetailResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SecretDetailResponseValidationError) ErrorName() string {
	return "SecretDetailResponseValidationError"
}

// Error satisfies the builtin error interface
func (e SecretDetailResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSecretDetailResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SecretDetailResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SecretDetailResponseValidationError{}

// Validate checks the field values on CreateSecretRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *CreateSecretRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreateSecretRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// CreateSecretRequestMultiError, or nil if none found.
func (m *CreateSecretRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *CreateSecretRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetClusterName()) < 1 {
		err := CreateSecretRequestValidationError{
			field:  "ClusterName",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetNamespace()) < 1 {
		err := CreateSecretRequestValidationError{
			field:  "Namespace",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetName()) < 1 {
		err := CreateSecretRequestValidationError{
			field:  "Name",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	// no validation rules for Labels

	// no validation rules for Annotations

	// no validation rules for Data

	if _, ok := SecretType_name[int32(m.GetType())]; !ok {
		err := CreateSecretRequestValidationError{
			field:  "Type",
			reason: "value must be one of the defined enum values",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	// no validation rules for TlsCrt

	// no validation rules for TlsKey

	// no validation rules for Registry

	// no validation rules for Username

	// no validation rules for Password

	if len(errors) > 0 {
		return CreateSecretRequestMultiError(errors)
	}

	return nil
}

// CreateSecretRequestMultiError is an error wrapping multiple validation
// errors returned by CreateSecretRequest.ValidateAll() if the designated
// constraints aren't met.
type CreateSecretRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreateSecretRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreateSecretRequestMultiError) AllErrors() []error { return m }

// CreateSecretRequestValidationError is the validation error returned by
// CreateSecretRequest.Validate if the designated constraints aren't met.
type CreateSecretRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreateSecretRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreateSecretRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreateSecretRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreateSecretRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreateSecretRequestValidationError) ErrorName() string {
	return "CreateSecretRequestValidationError"
}

// Error satisfies the builtin error interface
func (e CreateSecretRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreateSecretRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreateSecretRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreateSecretRequestValidationError{}

// Validate checks the field values on CreateSecretResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *CreateSecretResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on CreateSecretResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// CreateSecretResponseMultiError, or nil if none found.
func (m *CreateSecretResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *CreateSecretResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return CreateSecretResponseMultiError(errors)
	}

	return nil
}

// CreateSecretResponseMultiError is an error wrapping multiple validation
// errors returned by CreateSecretResponse.ValidateAll() if the designated
// constraints aren't met.
type CreateSecretResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m CreateSecretResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m CreateSecretResponseMultiError) AllErrors() []error { return m }

// CreateSecretResponseValidationError is the validation error returned by
// CreateSecretResponse.Validate if the designated constraints aren't met.
type CreateSecretResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CreateSecretResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CreateSecretResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CreateSecretResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CreateSecretResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CreateSecretResponseValidationError) ErrorName() string {
	return "CreateSecretResponseValidationError"
}

// Error satisfies the builtin error interface
func (e CreateSecretResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCreateSecretResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CreateSecretResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CreateSecretResponseValidationError{}

// Validate checks the field values on UpdateSecretRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *UpdateSecretRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on UpdateSecretRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// UpdateSecretRequestMultiError, or nil if none found.
func (m *UpdateSecretRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *UpdateSecretRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetClusterName()) < 1 {
		err := UpdateSecretRequestValidationError{
			field:  "ClusterName",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetNamespace()) < 1 {
		err := UpdateSecretRequestValidationError{
			field:  "Namespace",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetName()) < 1 {
		err := UpdateSecretRequestValidationError{
			field:  "Name",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	// no validation rules for Labels

	// no validation rules for Annotations

	// no validation rules for Data

	if len(errors) > 0 {
		return UpdateSecretRequestMultiError(errors)
	}

	return nil
}

// UpdateSecretRequestMultiError is an error wrapping multiple validation
// errors returned by UpdateSecretRequest.ValidateAll() if the designated
// constraints aren't met.
type UpdateSecretRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m UpdateSecretRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m UpdateSecretRequestMultiError) AllErrors() []error { return m }

// UpdateSecretRequestValidationError is the validation error returned by
// UpdateSecretRequest.Validate if the designated constraints aren't met.
type UpdateSecretRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UpdateSecretRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UpdateSecretRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UpdateSecretRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UpdateSecretRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UpdateSecretRequestValidationError) ErrorName() string {
	return "UpdateSecretRequestValidationError"
}

// Error satisfies the builtin error interface
func (e UpdateSecretRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUpdateSecretRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UpdateSecretRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UpdateSecretRequestValidationError{}

// Validate checks the field values on UpdateSecretResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *UpdateSecretResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on UpdateSecretResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// UpdateSecretResponseMultiError, or nil if none found.
func (m *UpdateSecretResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *UpdateSecretResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return UpdateSecretResponseMultiError(errors)
	}

	return nil
}

// UpdateSecretResponseMultiError is an error wrapping multiple validation
// errors returned by UpdateSecretResponse.ValidateAll() if the designated
// constraints aren't met.
type UpdateSecretResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m UpdateSecretResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m UpdateSecretResponseMultiError) AllErrors() []error { return m }

// UpdateSecretResponseValidationError is the validation error returned by
// UpdateSecretResponse.Validate if the designated constraints aren't met.
type UpdateSecretResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UpdateSecretResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UpdateSecretResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UpdateSecretResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UpdateSecretResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UpdateSecretResponseValidationError) ErrorName() string {
	return "UpdateSecretResponseValidationError"
}

// Error satisfies the builtin error interface
func (e UpdateSecretResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUpdateSecretResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UpdateSecretResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UpdateSecretResponseValidationError{}
